### DISCUSSION TITLE
RAM Tuning in Data Engine and Data Processor

### DISCUSSION THREAD

**Discussion by roberto103** [utmstack/UTMStack#655] [Q&A] [answered]
Hello, I have high RAM usage in the data engine and data processor, I tried to scale the node1 and logstash services from 1 to 2 in docker swarm, but it still didn't work.
I would like to know how to proceed, the host machine has 16GB of RAM.
![2024-04-30_10-58](https://github.com/utmstack/UTMStack/assets/42358389/dddf7ac0-d586-4f4e-a831-45ce4bbd409c)
![2024-04-30_10-57](https://github.com/utmstack/UTMStack/assets/42358389/7b94763a-87d9-4c13-9c85-a10adbbeb2f8)

**Comment by c3s4rfred** [comment]
Hi, @roberto103, the Data Processor and all the containers on version 10.4 has a self managed resources distribution, the installer calculates the resources distribution and asign to each container the proper amount of RAM. Some of the internal tools has a High RAM and/or CPU usage as CORRELATION, Data Processor, OpenSearch. You don't have to worry about that, it is a normal behaviour while the system don't get frezzed.

Best regards

**Comment by c3s4rfred** [comment, marked as answer]
Actually we don't perform horizontal scaling in those components but will do, is included in our road map. For now, you can increase the RAM to 32 and run the installer again.
Best regards
